test/test_utils.py

```diff
@@ -33,11 +33,22 @@
 
 
 class CmdStanPathTest(unittest.TestCase):
+
     def test_default_path(self):
-        abs_rel_path = os.path.expanduser(
-            os.path.join('~', '.cmdstanpy', 'cmdstan')
-        )
-        self.assertTrue(cmdstan_path().startswith(abs_rel_path))
+        if 'CMDSTAN' in os.environ:
+            self.assertEqual(cmdstan_path(), os.environ['CMDSTAN'])
+            path = os.environ['CMDSTAN']
+            del os.environ['CMDSTAN']
+            self.assertFalse('CMDSTAN' in os.environ)
+            set_cmdstan_path(path)
+            self.assertEqual(cmdstan_path(), path)
+            self.assertTrue('CMDSTAN' in os.environ)
+        else:
+            abs_rel_path = os.path.expanduser(
+                os.path.join('~', '.cmdstanpy', 'cmdstan')
+                )
+            self.assertTrue(cmdstan_path().startswith(abs_rel_path))
```

Comment:
Maybe use (os.abspath +) os.relpath comparison here?

relative path from abs_rel_path to env path should be empty (string) which should eval to False (bool), but you could also use explicit `==""`.

Reply:
I would prefer to leave this as is - is the objection to the "startswith" test?
relpath test didn't work for me, but samefile test did -
```
>>> foo = cmdstanpy.cmdstan_path()
>>> foo
'/Users/mitzi/.cmdstanpy/cmdstan-2.22.1'
>>> os.environ['CMDSTAN']
'/Users/mitzi/.cmdstanpy/cmdstan-2.22.1'
>>> os.path.relpath(os.environ['CMDSTAN'], foo)
'.'
>>> os.path.samefile(os.environ['CMDSTAN'], foo)
True
```

Reply:
yeah, startswith could fail when there are different mapping for the specific address.

samefile sounds totally fine

Reply:
samefile is too strict - this test as written is version agnostic - that's why it has to be "startswith"

Reply:
What do you mean by "this test as written is version agnostic"?

I think `samefile` just test if `relpath == "."` (not sure)

Reply:
the default path will include some version number - I need to change this test to use `utils.get_latest_cmdstan` - will do that, in which case this will work.

Reply:
nevermind - foggy brain - that's what the test is already doing - will change.

Reply:
OK, made changes per your suggestion - using `get_latest_version` to get full path to default cmdstan install.